Add new `require-prop-comment` rule (#1994)
vuejs/eslint-plugin-vue: docs/rules/require-prop-comment.md

---
pageClass: rule-details
sidebarDepth: 0
title: vue/require-prop-comment
description: require prop should have a comment
---
# vue/require-prop-comment

> require prop should have a comment

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>

## :book: Rule Details

> This rule requires that the prop should have a comment

Reviewer: ```suggestion
This rule enforces that every prop has a comment that documents it.
```

Author: It has been completed and will be merged in the next submission

<eslint-code-block :rules="{'vue/require-prop-comment': ['error']}">

```vue
<!-- ✓ GOOD -->
<template>
  <div>1</div>
</template>
<script>
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  props: {
    /**
     * a comment
     */
    a: Number,
    // b comment
    b: Number,
    // c
    // comment

>     c: Number

Reviewer: Should single-line comments really be considered "good"? When documenting props is the goal, I think JSDoc comments should be preferred over single-line comments.

Probably also documentation generator tools will ignore single-line comments. What's your opinion about this?

Author: I think JSDoc comments should be preferred over single-line comments too.
The second parameter will provide a choice，line 、block or unlimited
I am thinking about the following questions
1.Can I mix them?
2.Need to limit the quantity?

Reviewer: I think the rule should only consider JSDoc comments (`/** */`) and ignore block comments (`/* */`) and line comments `//`. No need to limit the number then, since only the JSDoc comment immediately before the statement will be considered.
